--- card-acos5.c.orig	2016-06-03 11:19:51.000000000 +0200
+++ card-acos5.c	2017-04-07 09:21:35.146697107 +0200
@@ -28,8 +28,6 @@
 #include "cardctl.h"
 
 static struct sc_atr_table acos5_atrs[] = {
-	{"3b:be:96:00:00:41:05:20:00:00:00:00:00:00:00:00:00:90:00", NULL, NULL,
-	SC_CARD_TYPE_ACOS5_GENERIC, 0, NULL},
 	{"3b:be:18:00:00:41:05:10:00:00:00:00:00:00:00:00:00:90:00", NULL, NULL,
 	 SC_CARD_TYPE_ACOS5_GENERIC, 0, NULL},
 	{NULL, NULL, NULL, 0, 0, NULL}
