--- $HOME/OpenSC/opensc/opensc-0.16.0_tar/src/pkcs11/framework-pkcs15.c	2016-06-04 01:22:08.000000000 +0200
+++ $HOME/trash/opensc-0.16.0/src/pkcs11/framework-pkcs15.c	2016-06-28 16:36:36.801809213 +0200
@@ -3589,7 +3589,7 @@
 	struct sc_pkcs11_card *p11card = session->slot->p11card;
 	struct pkcs15_fw_data *fw_data = NULL;
 	struct pkcs15_prkey_object *prkey;
-	unsigned char decrypted[256]; /* FIXME: Will not work for keys above 2048 bits */
+	unsigned char decrypted[512]; /* FIXME: Will not work for keys above 4096 bits */
 	int	buff_too_small, rv, flags = 0, prkey_has_path = 0;
 
 	sc_log(context, "Initiating decryption.");
